plugins/inputs/syslog/nontransparent_test.go

```diff
@@ -16,20 +16,25 @@ import (
 	"github.com/influxdata/telegraf/testutil"
 )
 
-func getTestCasesForNonTransparent() []testCaseStream {
+func getTestCasesForNonTransparent(hasRemoteAddr bool) []testCaseStream {
+	sourceAddr := ""
+	if hasRemoteAddr {
+		sourceAddr = "127.0.0.1"
+	}
+
```

Comment:
Can you please do this at the end of the definition (e.g. [line 140](https://github.com/influxdata/telegraf/blob/10b7c48d8ba744d1183e0e867ae6c7f7271e2e21/plugins/inputs/syslog/nontransparent_test.go#L140)) instead of here using e.g.

```golang
	if hasRemoteAddr {
		for i, tc := range testCases {
			for j, m := range tc.wantStrict {
				m.AddTag("source", "127.0.0.1")
				tc.wantStrict[j] = m
			}
			testCases[i] = tc
		}
	}
```

The same for the other tests below.

This has the advantage of less code and we can modify it in one place.

Reply:
Ah yeah, that's much better. Thank you! :)